### DISCUSSION TITLE
Uvicorn seemingly mixes up headers of different responses.

### DISCUSSION THREAD

**Discussion by re-masashi** [Kludex/uvicorn#1876] [Potential Issue]
I was creating an [example](https://github.com/re-masashi/example-willpyre) using an [ASGI framework](https://github.com/re-masashi/willpyre). The app was serving static assets on a separate route and the login page and other HTML responses on various different routes. 
Sometimes, when the page is reloaded multiple times, the HTML files are served with content type CSS (which was specified in the response to the previous request). 
I tried checking for any logical error in the source code of my app as well as the framework. There seem to be none that can cause this issue. 
To be sure that this is caused by Uvicorn, I tried reproducing the error with [Hypercorn](https://github.com/pgjones/hypercorn). The app works fine using Hypercorn. 

## Steps to reproduce

```bash 
git clone https://github.com/re-masashi/example-willpyre
cd example-willpyre
pip install -r requirements.txt
python -m uvicorn main:app # -- reload
```
Then visiting the page at http://127.0.0.0:8000/ and reloading multiple times.

## Machine specific info

OS: Arch Linux x86_64 
Host: 2539BD3 ThinkPad T410 
 Kernel: 6.1.12-arch1-1

**Comment by Kludex** [comment]
What are the headers that are mixed up?

**Reply by re-masashi** [reply]
Content -type, as we can see. I am not sure about other headers, it is possible that they can get mixed up.

**Reply by re-masashi** [reply]
The issue persists on Hypercorn with uvloop.

**Reply by re-masashi** [reply]
Update: The error goes away when I disable cache in Firefox. Had to reload two times after disabling cache to fox it. Further reloads with no caching gives proper responses. With caching enabled, the response is messed up despite  numerous reloads. Requesting with cURL gives proper results.

**Comment by re-masashi** [comment]
Can I kindly get any updates related to this? I checked the source code of [willpyre](https://github.com/re-masashi/willpyre) and it looks fine to me.
